cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -393,6 +419,24 @@ private void deployDefaultModules() {
     }
   }
 
+  private void deployExtensionModules() {
+    // adding any defined extension modules to be available in dataset manager service
+    for (Map.Entry<String, DatasetModule> module : extensionModules.entrySet()) {
+      try {
+        // NOTE: we assume extension modules are always in classpath, hence passing null for jar location
+        // NOTE: we add extension modules in the system namespace
+        Id.DatasetModule theModule = Id.DatasetModule.from(Id.Namespace.SYSTEM, module.getKey());
+        addModule(theModule, module.getValue().getClass().getName(), null);
+      } catch (DatasetModuleConflictException e) {
+        // perfectly fine: we need to add the modules only the very first time service is started
+        LOG.info("Not adding " + module.getKey() + " extension module: it already exists");
+      } catch (Throwable th) {
+        LOG.error("Failed to add {} extension module. Aborting.", module.getKey(), th);
```

Comment:
This pattern (log and re-throw) followed the previously existing pattern in DeployDefaultModules above this routine and in fact the log/throw pattern is used several places in this file, should this be changed? If a message is desired to be added in addition to the exception in these cases what is the suggested pattern (e.g., something like throw new RuntimeException("msg, exception: "+exc.toString())? thx